examples/next-prisma-starter/src/pages/index.tsx

```diff
@@ -18,6 +18,13 @@ const IndexPage: NextPageWithLayout = () => {
     },
   );
 
+  utils.post.list.setData((old) => {
+    old;
+    //^?
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return old!;
```

Comment:
you should be able to return `old` without the non-null assertion, because returning `undefined` from `setQueryData` is allowed in v4. We'll only create the cache entry if you return something that is not `undefined`, which is also the reason why `undefined` is disallowed a value for a successful query.

```suggestion
    return old;
```

Reply:
Interesting, might have to look into the types then cause I get typeerrors if i return undefined 🤔 

<img width="536" alt="CleanShot 2022-09-21 at 16 02 43@2x" src="https://user-images.githubusercontent.com/51714798/191525509-d0df0a1d-109f-4054-9ddb-59907ffb45b3.png">

Reply:
Should we no longer have the `| undefined` then?
<img width="562" alt="CleanShot 2022-09-21 at 16 06 22@2x" src="https://user-images.githubusercontent.com/51714798/191526303-47631cb7-154e-4400-a28c-a12d6d859281.png">

Reply:
agree, it should be:

`Updater<TOutput | undefined, TOutput | undefined>`

see: https://github.com/TanStack/query/blob/454c58481f6f5d5c3e182e6e418d65fdb4a9ec1f/packages/query-core/src/queryClient.ts#L129-L133